modules/core/server/controllers/mongoose-filtering.server.controller.js

```diff
@@ -0,0 +1,54 @@
+﻿// Service for managing Filtering/Parameterized-Queries.
+
+var _query = null;
+var _modifiedQuery = null;
+var _totalDocumentCount = null;
+var _page = 0;
+var _take = 50; // default to prevent unreasonable take() requests
```

Comment:
I think it makes sense to make *take* and *page* query-string parameters. It's how I've seen a lot of other websites do it, because it allows people to (at least for a short period of time) return to a similar place in a results feed rather than having to start from page 1 again when returning to the old URL.